manifests/maintenance/pg_repack.pp

```diff
@@ -0,0 +1,59 @@
+# A description of what this class does
+#
+# @summary A short summary of the purpose of this class
+#
+# @example
+#   include pe_databases::maintenance::pg_repack
+class pe_databases::maintenance::pg_repack (
+  Boolean $disable_maintenance = $pe_databases::maintenance::disable_maintenance,
+  String  $logging_directory   = $pe_databases::maintenance::logging_directory,
+) {
+
+  $ensure_cron = $disable_maintenance ? {
+    true    => absent,
+    default => present
+  }
+
+  $repack          = 'su - pe-postgres -s /bin/bash -c "/opt/puppetlabs/server/apps/postgresql/bin/pg_repack -d pe-puppetdb'
+  $facts_tables    = '-t factsets -t fact_paths"'
+  $catalogs_tables = '-t catalogs -t catalog_resources -t edges -t certnames"'
+  $other_tables    = '-t producers -t resource_params -t resource_params_cache"'
+  $reports_tables  = '-t reports"'
+  $logging         = "> ${logging_directory}/output.log 2>&1"
```

Comment:
Is there any logic to ensure that multiple repacks cannot be started simultaneously? That's the biggest danger of automated repacking.

Reply:
The only ones that could overlap are the other tables and the reports table which run at 5:30 on 1 day of the month while facts and catalogs run at 4:30 2 days a week.  

I'm not particularly concerned about them overlapping... but if we thought it was a serious concern then we could move the times they run further apart.